test/apps/data-filter/src/data-filter.js

```diff
@@ -0,0 +1,42 @@
+/*
+ * data filter shader module
+ */
+const vs = `
+uniform vec2 filter_range;
+varying float filter_isVisible;
+
+void filter_setVisibility(float value) {
+  filter_isVisible = step(filter_range.x, value) * step(value, filter_range.y);
+}
+void filter_setVisibility(bool visible) {
+  filter_isVisible = float(visible);
+}
+`;
+
+const fs = `
+varying float filter_isVisible;
+vec4 filter_filterColor(vec4 color) {
+  if (filter_isVisible < 0.5) {
+    discard;
+  }
+  return color;
+}
+`
+
+const INITIAL_MODULE_OPTIONS = {};
+const DEFAULT_FILTER_RANGE = [0, 2];
+
+export default {
+  name: 'data-filter',
+  vs,
+  fs,
+  getUniforms: (opts = INITIAL_MODULE_OPTIONS) => {
+    if (!opts.data) {
```

Comment:
This looks like an undesirable knowledge of the layer system?

Why is it needed? 

Ideally, shader modules should stand alone.

Reply:
The auto highlighting system calls `updateModuleSettings` with the picked color. In which case no other layer props are present.